tests/integration/domains/osu_test.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+
+import hashlib
+import secrets
+from datetime import datetime
+from uuid import UUID
+
+from fastapi import status
+from httpx import AsyncClient
+
+from app import encryption
+
+
+async def test_score_submission(http_client: AsyncClient) -> None:
```

Comment:
This is currently very low-level in implementation. I think I'll defer the extraction/abstraction work to a future PR, as it will be more clear what should be extracted once there are a couple of tests.